front/src/views/Register.vue

```diff
@@ -248,11 +255,34 @@
                     }
                 }
             }, 500);
+
+            this.debouncedEmailSearch = debounce(() => {
+                if (this.email && this.email !== "") {
+                    if (/.+@.+\..+/.test(this.email)) {
+                        axios.post('/user/exists/email', {email: this.email}).then(res => {
+                            this.emailLoading = false;
+                            this.emailValid = !res.data.exists;
+                            if (res.data.exists) {
+                                this.emailError = 'That email is in use!';
+                            }
+                        }).catch(() => {
+                            this.emailLoading = false;
+                            this.emailError = 'Something went wrong!';
+                        })
+                    } else {
+                        this.emailLoading = false;
+                        this.emailError = 'Invalid Email';
+                    }
+                } else {
+                    this.emailLoading = false;
+                    this.emailValid = false;
+                    this.emailError = 'Email is required';
+                }
```

Comment:
```suggestion
                    this.emailLoading = false;
                }
```
Put this at the end and remove all `this.emailLoading = false` from every statement.

Reply:
this has a request through axios which is async so this wouldn't accurately show the loading